src/services/fhirService.ts

```diff
@@ -425,11 +425,9 @@ const fhirService = {
   ): Promise<IRecord> {
     let validationResult;
     try {
-      validationResult = await fhirValidator.validate(fhirResource);
+      validationResult = await fhirValidator.validate(cleanResource(cloneDeep(fhirResource)));
```

Comment:
What do you think about moving this to right before handing it to the recordservice?
It's great that we trust ourselves, but we could make sure that we don't break the esource on the way.